pyne/mesh.py

```diff
@@ -583,6 +583,47 @@ def expand(self):
             self.mesh.mesh.tag_set_data(tag, list(self.mesh.iter_ve()), data)
 
 
+    def mult(self, multiplier):
+        """Multiplication operator for NativeMeshTag. Multiplies self by the multiplier using 
+        NumPy slicing if applicable. Multiplier can be any of the following types: int, float, 
+        ndarray, list, NativeMeshTag. Throws error if shapes are incorrect """
+
+        if isinstance(multiplier, NativeMeshTag):
+            if self.size == 1:
+                if multiplier.size == 1:
+                    try:
+                        return self[:] * multiplier[:]
+                    except:
+                        print("Scalar data must have the same shape")
+                        #raise e 
+                elif multiplier.size > 1:
+                    try:
+                        return self[:][:, None] * multiplier[:]
+                    except:
+                        print("Incompatible shape for scalar or vector data")
+                        #raise e
+            elif self.size > 1:
+                if multiplier.size == 1:
+                    try:
+                        return self[:] * multiplier[:][None, :]
+                    except:
+                        print("Incompatible shape for vector or scalar data")
+                        #raise e
+                elif multiplier.size > 1:
+                    try:
+                        return self[:] * multiplier[:]
+                    except:
+                        print("Vectors must have the same shape")
+                        #raise e
+        elif isinstance(multiplier, int) or isinstance(multiplier, float):
+            return self[:]*multiplier
+        elif isinstance(multiplier, np.ndarray) or isinstance(multiplier, list):
+            try:
+                return self[:]*multiplier
+            except:
+                print("Incompatible vector multiplication")
+                #raise e
+
```

Comment:
suggestion: Add an `else` statement that throws a `TypeError`, e.g. if the user gives a string